hack/local-up-cluster.sh

```diff
@@ -688,111 +687,113 @@ function start_kubelet {
     if [[ -n "${CONTAINER_RUNTIME_ENDPOINT}" ]]; then
-      container_runtime_endpoint_args="--container-runtime-endpoint=${CONTAINER_RUNTIME_ENDPOINT}"
+      container_runtime_endpoint_args=("--container-runtime-endpoint=${CONTAINER_RUNTIME_ENDPOINT}")
     fi
 
-    image_service_endpoint_args=""
+    image_service_endpoint_args=()
     if [[ -n "${IMAGE_SERVICE_ENDPOINT}" ]]; then
-      image_service_endpoint_args="--image-service-endpoint=${IMAGE_SERVICE_ENDPOINT}"
+      image_service_endpoint_args=("--image-service-endpoint=${IMAGE_SERVICE_ENDPOINT}")
     fi
 
+    # shellcheck disable=SC2206
     all_kubelet_flags=(
-      ${priv_arg}
-      --v="${LOG_LEVEL}"
-      --vmodule="${LOG_SPEC}"
-      --chaos-chance="${CHAOS_CHANCE}"
-      --container-runtime="${CONTAINER_RUNTIME}"
-      --hostname-override="${HOSTNAME_OVERRIDE}"
-      ${cloud_config_arg}
-      --address="${KUBELET_HOST}"
+      "${priv_arg}"
+      "--v=${LOG_LEVEL}"
+      "--vmodule=${LOG_SPEC}"
+      "--chaos-chance=${CHAOS_CHANCE}"
+      "--container-runtime=${CONTAINER_RUNTIME}"
+      "--hostname-override=${HOSTNAME_OVERRIDE}"
+      "${cloud_config_arg[@]}"
+      "--address=${KUBELET_HOST}"
       --kubeconfig "${CERT_DIR}"/kubelet.kubeconfig
-      --feature-gates="${FEATURE_GATES}"
-      --cpu-cfs-quota="${CPU_CFS_QUOTA}"
-      --enable-controller-attach-detach="${ENABLE_CONTROLLER_ATTACH_DETACH}"
-      --cgroups-per-qos="${CGROUPS_PER_QOS}"
-      --cgroup-driver="${CGROUP_DRIVER}"
-      --cgroup-root="${CGROUP_ROOT}"
-      --eviction-hard="${EVICTION_HARD}"
-      --eviction-soft="${EVICTION_SOFT}"
-      --eviction-pressure-transition-period="${EVICTION_PRESSURE_TRANSITION_PERIOD}"
-      --pod-manifest-path="${POD_MANIFEST_PATH}"
-      --fail-swap-on="${FAIL_SWAP_ON}"
-      ${auth_args}
-      ${dns_args}
-      ${cni_conf_dir_args}
-      ${cni_bin_dir_args}
-      ${net_plugin_args}
-      ${container_runtime_endpoint_args}
-      ${image_service_endpoint_args}
-      --port="${KUBELET_PORT}"
+      "--feature-gates=${FEATURE_GATES}"
+      "--cpu-cfs-quota=${CPU_CFS_QUOTA}"
+      "--enable-controller-attach-detach=${ENABLE_CONTROLLER_ATTACH_DETACH}"
+      "--cgroups-per-qos=${CGROUPS_PER_QOS}"
+      "--cgroup-driver=${CGROUP_DRIVER}"
+      "--cgroup-root=${CGROUP_ROOT}"
+      "--eviction-hard=${EVICTION_HARD}"
+      "--eviction-soft=${EVICTION_SOFT}"
+      "--eviction-pressure-transition-period=${EVICTION_PRESSURE_TRANSITION_PERIOD}"
+      "--pod-manifest-path=${POD_MANIFEST_PATH}"
+      "--fail-swap-on=${FAIL_SWAP_ON}"
+      "${auth_args[@]}"
+      "${dns_args[@]}"
+      "${cni_conf_dir_args[@]}"
+      "${cni_bin_dir_args[@]}"
+      "${net_plugin_args[@]}"
+      "${container_runtime_endpoint_args[@]}"
+      "${image_service_endpoint_args[@]}"
+      "--port=${KUBELET_PORT}"
       ${KUBELET_FLAGS}
     )
 
     if [[ "${REUSE_CERTS}" != true ]]; then
         generate_kubelet_certs
     fi
 
+    # shellcheck disable=SC2024
```

Comment:
I think it's a redirect to the normal user.
So I ignore this message.

In ./hack/local-up-cluster.sh line 788:
    sudo -E "${GO_OUT}/hyperkube" kubelet "${all_kubelet_flags[@]}" >"${KUBELET_LOG}" 2>&1 &
                                                                    ^-- SC2024: sudo doesn't affect redirects. Use ..| sudo tee file